### PR TITLE
Restructure of the overlay agent module.

### PR DESCRIPTION
**Pull request by asridharan**
* Restructured the overlay agent module so that we can expose the `agent::ManagerProcess` in unit-tests.
* Re-wrote the unit-tests to use `agent::ManagerProcess`.

### CODE REVIEW

**Comment by asridharan** [comment]
@kaysoky this is the re-based code for re-organizing the agent overlay module. Could you kindly review it.

**Comment by asridharan** [comment]
@jieyu these commits should break the dependency of the agent start on the overlay module. Will be adding a test to check the problem with `configAttempts` not being reset on master failover.

File: overlay/agent.hpp [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+  Manager(const process::Owned<ManagerProcess>& _process);
+
+  process::Owned<ManagerProcess> process;
+};
+
+} // namespace agent {
+} // namespace overlay{
+} // namespace modules {
+} // namespace mesos {
+#endif
```
Comment (kaysoky) [inline overlay/agent.hpp]: `// __AGENT_OVERLAY_MANAGER_HPP__`

Comment (kaysoky) [inline overlay/agent.hpp]: And add a newline before.

File: overlay/agent.hpp [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+  virtual ~Manager();
+
+private:
+  Manager(const process::Owned<ManagerProcess>& _process);
+
+  process::Owned<ManagerProcess> process;
+};
+
+} // namespace agent {
+} // namespace overlay{
```
Comment (kaysoky) [inline overlay/agent.hpp]: Missing space: `overlay {`

File: overlay/agent.hpp [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+#include <mesos/mesos.hpp>
+#include <mesos/module/anonymous.hpp>
+
+#include <overlay/messages.hpp>
+
+namespace mesos {
+namespace modules {
+namespace overlay {
+namespace agent {
+class ManagerProcess : public ProtobufProcess<ManagerProcess>
```
Comment (kaysoky) [inline overlay/agent.hpp]: Newline before this.

**Pull request merged**